Plugin.pm

```diff
@@ -954,31 +959,74 @@ sub QobuzGetTracks {
 					image => $formattedTrack->{image},
 					tracks => []
 				} unless $works->{$workId};
-
+				
+				if ($workId ne $lastwork) {  # create a new work heading
+#					if ($i && $lastwork eq "") {  # create a separator line after previous single track#
+#						push @$items,{
+#							name  => "--------",
+#							type  => 'text'
+#						};
+#					}
+					push @$items,{
+						name  => $work,
+						type  => 'text'
+					};
+						
+					$lastwork = $workId;
+				}
+				
 				push @{$works->{$workId}->{tracks}}, $formattedTrack;
-			}
+				if (scalar @{$works->{$workId}->{tracks}} > 1) {
+					$worksfound = 1;   # we found a work with more than one track
+				}
+			} elsif ($lastwork ne "") {  # create a separator line for tracks without a work
+						push @$items,{
+							name  => "--------",
+							type  => 'text'
+						};
+						
+					$lastwork = "";
+			}	
 
 			$i++;
 
 			push @$items, $formattedTrack;
 		}
 
 		if (scalar keys %$works) {
-			foreach my $work (sort { $works->{$b}->{index} <=> $works->{$a}->{index} } keys %$works) {
-				my $workTracks = $works->{$work}->{tracks};
-
-				# insert works item before the first of its tracks
-				splice @$items, $works->{$work}->{index}, 0, {
-					name => $works->{$work}->{title},
-					image => $works->{$work}->{image},
-					type => 'playlist',
-					playall => 1,
-					url => \&QobuzWorkGetTracks,
-					passthrough => [{
-						tracks => $workTracks
-					}],
-					items => $workTracks
-				} if scalar @$workTracks > 1;
+			if ( $prefs->get('workPlaylistPosition') eq "before" ) {
+				# insert work playlists before the tracks unless there is only one work containing all tracks
+				foreach my $work (sort { $works->{$b}->{index} <=> $works->{$a}->{index} } keys %$works) {
+					my $workTracks = $works->{$work}->{tracks};
+					splice @$items, 0, 0, {
+						name => $works->{$work}->{title},
+						image => 'html/images/playlists.png',
+						type => 'playlist',
+						playall => 1,
+						url => \&QobuzWorkGetTracks,
+						passthrough => [{
+							tracks => $workTracks
+						}],
+						items => $workTracks
+					} if $worksfound && scalar @$workTracks && scalar @$workTracks < $album->{tracks_count};
```

Comment:
A question of style... but if I have such a longish instruction and a longish condition, then I prefer to have a regular `if (cond) { block of code... }` statement.